packages/core/src/typeschema/validation.ts

```diff
@@ -149,10 +157,22 @@ class ResourceValidator {
         );
       }
 
-      this.checkSpecifiedValue(value, element, path);
+      if (!matchesSpecifiedValue(value, element)) {
+        this.issues.push(createStructureIssue(path, 'Value did not match expected pattern'));
+      }
+      const sliceCounts: Record<string, number> = Object.fromEntries(
```

Comment:
I see at least 3 different checks for `element.slicing` - I wonder if you could do a minor refactor to do that check once to simplify

```ts
const slices = element.slicing?.slices;
const sliceCounts = slices.reduce((acc, s) => ({ ...acc, [s.name]: 0 }), {}) ?? [];

// ...

if (slices) {
  // ...
}
```

Using `?? []` makes for shorter code, but it leads to unnecessary allocations and unnecessary function calls which we may want to avoid in an inner inner loop like this.

Reply:
Thanks — that's a great callout on the allocation!  I can definitely refactor that out